### PR TITLE
react router

### PR DESCRIPTION
**Pull request by moog16**
[x] routing
[x] decreased initial load time
[x] created JSX pages in place of index.html boilerplate
[x] css modules for css obfuscation - there will be no class name collisions on a per component basis

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,6 +42,9 @@
     "eslint-plugin-react": "^7.7.0",
     "extract-text-webpack-plugin": "^3.0.2",
     "fs-readfile-promise": "^3.0.1",
+    "handlebars": "^4.0.11",
```
Comment (lynnjepsen) [inline package.json]: handlebars...wait wha?

File: package.json [hunk 2]
```diff
@@ -62,5 +65,8 @@
     "validate-commit-msg": "^2.14.0",
     "webpack": "^3.11.0",
     "webpack-dev-server": "^2.11.2"
+  },
+  "dependencies": {
+    "react-router-dom": "^4.2.2"
```
Comment (lynnjepsen) [inline package.json]: this should be a dev dependency too. None of our shipped NPM modules will depend on react-router...just our developer workflow

File: test/screenshot/App.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+class RouteContainer_ extends React.Component {
+  render() {
+    return (
+      <main>
+        <nav className="mdc-toolbar-fixed-adjust">
+          <Switch>
+            <Route exact path="/" component={Catalog} />
+            <Route exact path="/temporary-package" component={TempPackage} />
+            <Route path="/temporary-package/foo-test" component={FooTest} />
```
Comment (lynnjepsen) [inline test/screenshot/App.js]: Im a little afraid of merge conflicts in this big test/screenshot/App.js file. Can you isolate the /temporary-packages logic into a file in the actual temporary-package directory

File: test/screenshot/screenshot.js [hunk 4]
```diff
@@ -53,6 +53,7 @@ export default class Screenshot {
     const browser = await puppeteer.launch();
     const page = await browser.newPage();
     await page.goto(url);
+    console.log(path);
```
Comment (lynnjepsen) [inline test/screenshot/screenshot.js]: hmmmmmmmm

File: test/screenshot/index.hbs [hunk 5]
```diff
@@ -1,15 +1,16 @@
-<!DOCTYPE html>
+<!doctype html>
```
Comment (lynnjepsen) [inline test/screenshot/index.hbs]: does this have to be a handlebars file? Can we just serve a straight HTML file and be done?

File: test/screenshot/webpack-bundles.js [hunk 6]
```diff
@@ -42,8 +47,14 @@ module.exports.bundle = function(testPath, outputPath) {
...
     plugins: [
-      new ExtractTextPlugin(outputPath + '.css'),
+      new ExtractTextPlugin('bundle.css'),
+      new HtmlWebpackPlugin({
+        template: path.resolve(__dirname, 'index.hbs'),
+      }),
       new OptimizeCssAssetsPlugin(),
     ],
+    devServer: {
+      historyApiFallback: true,
```
Comment (lynnjepsen) [inline test/screenshot/webpack-bundles.js]: but why?

File: test/screenshot/webpack.config.js [hunk 7]
```diff
@@ -1,6 +1,3 @@
-const temporaryPackageBundles
-  = require('./temporary-package/webpack.config.js');
+const config = require('./webpack-bundles');
```
Comment (lynnjepsen) [inline test/screenshot/webpack.config.js]: This file doesnt make sense anymore

File: test/screenshot/App.js (hunk 3 above)
Comment (moog16) [reply test/screenshot/App.js]: Ya I was also thinking that. Sounds good

File: test/screenshot/webpack-bundles.js (hunk 6 above)
Comment (moog16) [reply test/screenshot/webpack-bundles.js]: This is how the dev server knows to serve the single page app again. Say we go to `temp-package/foo` path. That is not an html file, so we serve the index.html file, which hits the JS bundle. The react router then routes to the correct JS to serve based on the location.

**Comment by moog16** [comment]
@lynnjepsen added css modules for styles obfuscation. This allows us to use the define the class names without having collision from other stylesheets. It also enables us to continue to use our existing styles from MDC Web.

File: package.json [hunk 8]
```diff
@@ -60,9 +62,12 @@
     "mocha": "^5.0.2",
     "node-sass": "^4.7.2",
     "optimize-css-assets-webpack-plugin": "^3.2.0",
+    "postcss-icss-selectors": "^2.0.3",
```
Comment (moog16) [inline package.json]: needed for css-modules

File: test/screenshot/material-icon/index.scss [hunk 9]
```diff
@@ -1,3 +1,3 @@
-body {
-  margin: 24px;
+:global {
```
Comment (moog16) [inline test/screenshot/material-icon/index.scss]: If anything is imported from the packages directory, wrap it in `:global` so it isn't obfuscated.

File: test/screenshot/top-app-bar/standard.js [hunk 10]
```diff
@@ -1,18 +1,20 @@
...
 import TopAppBar from '../../../packages/top-app-bar';
-import '../../../packages/top-app-bar/index.scss';
-import '../../../packages/material-icon/index.scss';
-import './index.scss';
-
 import MaterialIcon from '../../../packages/material-icon';
+import classnames from 'classnames/bind';
+
+import styles from './test1.scss';
+const cx = classnames.bind(styles);
```
Comment (moog16) [inline test/screenshot/top-app-bar/standard.js]: if styles are obfuscated via css modules, them in this `bind` method to get a mapping of it. Then we can use it below:

```
cx('demo') // gets the obfuscated class name
```

**Comment by moog16** [comment]
Closing this since its gone stale

**Pull request closed without merging**